From 476ab2af3d91a3bc16ec588d57d38c06db4efb58 Mon Sep 17 00:00:00 2001
From: "Valery V. Vorotyntsev" <valery_vorotyntsev@xyratex.com>
Date: Thu, 22 Aug 2013 16:10:40 +0300
Subject: [PATCH 7/7] ut: fix m0_ut_be_tx_begin()

sm_group is not locked automatically any more.
This change lets fol-ut:fol-init pass.

Kudos to Maxim Medved who helped to solve this issue.
---
 ut/be.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/ut/be.c b/ut/be.c
index 531005b..e230773 100644
--- a/ut/be.c
+++ b/ut/be.c
@@ -51,11 +51,11 @@ M0_INTERNAL void m0_ut_be_tx_begin(struct m0_be_tx *tx,
 				   struct m0_be_ut_backend *ut_be,
 				   struct m0_be_tx_credit *cred)
 {
-	extern struct m0_sm_group ut__txs_sm_group;
 	int rc;
 
-	m0_be_tx_init(tx, 0, &ut_be->but_dom, &ut__txs_sm_group, NULL, NULL,
-		      NULL, NULL);
+	m0_be_tx_init(tx, 0, &ut_be->but_dom,
+		      m0_be_ut_backend_sm_group_lookup(ut_be),
+		      NULL, NULL, NULL, NULL);
 	m0_be_tx_prep(tx, cred);
 	m0_be_tx_open(tx);
 	rc = m0_be_tx_timedwait(tx, M0_BITS(M0_BTS_ACTIVE), M0_TIME_NEVER);
-- 
1.8.3.2

